### ISSUE TITLE
Duplicate docs backfilled (cc) when issuing a longpoll changes after a backfill

### ISSUE THREAD

**Issue by sethrosetter** [couchbase/sync_gateway#2277]
### Sync Gateway version
1.4-55 

### Operating system
CentOS

### Config file
```
{
    "interface":":4984",
    "maxIncomingConnections": 0,
    "maxCouchbaseConnections": 16,
    "maxFileDescriptors": 90000,
    "slowServerCallWarningThreshold": 500,
    "compressResponses": false,
    "log": ["CRUD+", "Cache+", "HTTP+", "Changes+"],
    "verbose":"true",
    "databases":{
        "db":{
            "server":"http://{{ couchbase_server_primary_node }}:8091",
            "bucket":"data-bucket",
            "sync":
            `function(doc, oldDoc){
                if(doc._id == "channel_access") {
                    access(doc.users, doc.channels);
                } else if(doc._id == "role_access") {
                    role(doc.users, doc.roles);
                } else if(doc._id == "channel_grant_to_role") {
                    access(doc.roles, doc.channels);
                } else {
                    channel(doc, doc.channels);
                }
            }`
        }
    }
}
```

### Steps to reproduce
1.  Create a user with a role with no channels ("empty_role")
2.  Create 50 docs with channel ("A")
3.  Grant "empty_role" access to channel "A", with update to role via REST or access() via sync
4.  Issue a longpoll changes request (changes 1)
5. Issue a longpoll changes request since=changes 1's last_seq

### Expected behavior
You should see no changes

### Actual behavior
You see all the changes from the previous longpoll

**Comment by sethrosetter** [comment]
When this gets fixed, we need to update the tests - https://github.com/couchbaselabs/mobile-testkit/blob/master/testsuites/syncgateway/functional/tests/test_changes_backfill.py#L800-L810

**Comment by adamcfraser** [comment]
When this gets fixed, should take into account #2258

**Comment by adamcfraser** [comment]
Closing based on age.
